DataFormats/ParticleFlowReco/interface/PFRecHit.h

```diff
@@ -126,14 +126,14 @@ namespace reco {
 
 
     /// rechit cell centre x, y, z
-    PositionType const & position() const { return caloCell().getPosition().basicVector(); }
+    PositionType position() const { return caloCell().getPosition().basicVector(); }
```

Comment:
why were these made to return a copy?
the position is probably small enough that a copy is fine (being inlined), but the CornersVec copies seem too much.

The lifetime of this is as long as the PFRecHit itself. So, making a ref should be OK.

Reply:
I agree with Slava.